src/base64.cr

```diff
@@ -123,7 +123,7 @@ module Base64
     count = 0
     to_base64(data.to_slice, alphabet, pad: pad) do |byte|
       count += 1
-      io.write_byte byte
+      io << byte.unsafe_chr
     end
```

Comment:
there is also decode write_byte: https://github.com/crystal-lang/crystal/blob/ee5c8b6b345c98abbbcffbee09d09295d8f120b9/src/base64.cr#L174

Reply:
`Base64.decode` produces raw binary data, so `byte` does not represent an ASCII character here, meanwhile `decode_string` simply casts the bytes into a `String` without regard for UTF-8 validity.